pkg/api/v1alpha1/edgeconnect/edgeconnect_types.go

```diff
@@ -91,6 +99,26 @@ type ImageRefSpec struct {
 	Tag string `json:"tag,omitempty"`
 }
 
+type ProxySpec struct {
+	// Proxy scheme.
+	Scheme string `json:"scheme,omitempty"`
+
+	// Server address (hostname or IP address) of the proxy.
+	Host string `json:"host,omitempty"`
+
+	// NoProxy represents the NO_PROXY or no_proxy environment
+	// variable. It specifies a string that contains comma-separated values
+	// specifying hosts that should be excluded from proxying.
+	NoProxy string `json:"noProxy,omitempty"`
+
+	// Secret name which represents username and password for authentication with the proxy, using the
+	// "Basic" HTTP authentication scheme.
```

Comment:
```suggestion
	// Secret name which contains the username and password used for authentication with the proxy, using the
	// "Basic" HTTP authentication scheme.
```

We could also add which "keys" should be used in the secret. (if we already decided)